src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -140,9 +143,21 @@ public static class ODataBatchReaderExtensions
 
             foreach (var header in batchRequest.Headers)
             {
-                // Copy headers from batch, overwriting any existing headers.
                 string headerName = header.Key;
                 string headerValue = header.Value;
+
+                if (headerName.Trim().ToLowerInvariant() == "prefer")
```

Comment:
Could invert the ```if``` to take care of the ```request.Headers[headerName] = headerValue;``` line then leave the rest of the code un-indented...

Reply:
In this case, since the prefer header is sort of an edge case, I think it makes sense to handle the edge case, rather than the default case, in the if block.